ipaserver/rpcserver.py

```diff
@@ -493,7 +493,7 @@ def marshal(self, result, error, _id=None,
 
     def unmarshal(self, data):
         try:
-            d = json.loads(data)
+            d = json.loads(ipautil.decode_json(data))
```

Comment:
We currently support only UTF-8:
```
$ echo '{"method": "ping", "params": [[], {}]}' | iconv -f utf-8 -t utf-16 | curl https://$HOSTNAME/ipa/json --cacert /etc/ipa/ca.crt --negotiate -u : -e https://$HOSTNAME/ipa/json -H 'Content-Type: application/json' --data-binary @-
{
    "error": {
        "code": 909, 
        "data": {
            "error": "No JSON object could be decoded"
        }, 
        "message": "Invalid JSON-RPC request: No JSON object could be decoded", 
        "name": "JSONError"
    }, 
    "id": null, 
    "principal": "admin@ABC.IDM.LAB.ENG.BRQ.REDHAT.COM", 
    "result": null, 
    "version": "4.4.90.dev201701171151+git7aee0d4"
}
```
Adding support for other encodings is not in the scope of this effort, and should be based on HTTP headers rather than guesswork anyway.